### PR TITLE
Include usage stats

### PR DESCRIPTION
**Pull request by ronaldmannak**
OpenAI [added usage stats to streaming chat completions](https://twitter.com/OpenAIDevs/status/1787573348496773423) today. This PR adds a new `usage` stats property to StreamableChatThread that will be set if `completion` is called with argument `includeUsage` to true.

Additionally, I've added a `completeIncludeUsage` method to the async `ChatThread` that returns a tuple of the completion and the usage.

### CODE REVIEW

File: Sources/CleverBird/chat/streaming/StreamOptions.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//
```
Comment (btfranklin) [inline Sources/CleverBird/chat/streaming/StreamOptions.swift:1]: Please remove this comment header. These comment headers that repeat the name of the file are just noise and don't add any value to the code.

File: Sources/CleverBird/chat/streaming/StreamOptions.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//  StreamOptions.swift
+//
+//
+//  Created by Ronald Mannak on 5/6/24.
```
Comment (btfranklin) [inline Sources/CleverBird/chat/streaming/StreamOptions.swift:5]: Feel free to leave this Created by line though. :)

**Review by btfranklin — CHANGES_REQUESTED** [review]
Just one minor change to a comment header, please. Otherwise, looking great!

File: Sources/CleverBird/chat/streaming/StreamOptions.swift (hunk 2 above)
Comment (ronaldmannak) [reply Sources/CleverBird/chat/streaming/StreamOptions.swift:1]: @btfranklin Done!

**Review by btfranklin — APPROVED** [review]

**Pull request merged**